File: C++/프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <iostream>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    
    int cur_idx;
    for(auto& c : s) {
        cur_idx = 0;
        c = c - 97;
        while(cur_idx < index) {
            c = ++c % 26;
            cur_idx += int(skip.find(char(c + 97)) == -1);
        }
        answer += char(c + 97);
    }
    
    return answer;
}

File: Python3/프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer, cur_idx = '', 0
    skip = [ord(sk) - 97 for sk in skip]
    s = [ord(c) - 97 for c in s]
    
    cur_idx = 0
    for c in s:
        cur_idx = 0
        while cur_idx < index:
            c = (c + 1) % 26
            cur_idx += int(c not in skip)
        answer += chr(c + 97) 
    
    return answer